sycl/doc/extensions/supported/sycl_ext_intel_dataflow_pipes.asciidoc

```diff
@@ -642,55 +642,47 @@ Automated mechanisms are possible to provide uniquification across calls, and co
-* `sycl::ext::intel::experimental::latency_constraint<A, B, C>`: A tuple of three values which cause the current read/write function call to act as an endpoint of a latency constraint relative to a specified `latency_anchor_id` defined by a different instruction.
+* `sycl::ext::oneapi::experimental::latency_anchor_id<N>`, where `N` is an integer: An ID to associate with the current read/write function call, which can then be referenced by other `latency_constraint` properties elsewhere in the program to define relative latency constaints. ID must be unique within the application, and a diagnostic is required if that condition is not met.
+* `sycl::ext::oneapi::experimental::latency_constraint<A, B, C>`: A tuple of three values which cause the current read/write function call to act as an endpoint of a latency constraint relative to a specified `latency_anchor_id` defined by a different instruction.
 ** `A` is an integer: The ID of the target anchor defined on a different instruction through a `latency_anchor_id` property.
-** `B` is an enum value: The type of control from the set {`type::exact`, `type::max`, `type::min`}.
+** `B` is an enum value: The type of control from the set {`latency_control_type::exact`, `latency_control_type::max`, `latency_control_type::min`}.
 ** `C` is an integer: The relative clock cycle difference between the target anchor and the current function call, that the constraint should infer subject to the type of the control (exact, max, min).
 
-The template arguments above don't have to be specified if user doesn't want to apply latency controls. The template arguments can be passed in arbitrary order.
-
 === Implementation
 
 [source,c++]
 ----
-// Added in version 2 of this extension.
-namespace sycl::ext::intel::experimental {
-enum class type {
-  none, // default
-  exact,
-  max,
-  min
-};
-
-template <int32_t _N> struct latency_anchor_id {
-  static constexpr int32_t value = _N;
-  static constexpr int32_t default_value = -1;
-};
-
-template <int32_t _N1, type _N2, int32_t _N3> struct latency_constraint {
-  static constexpr std::tuple<int32_t, type, int32_t> value = {_N1, _N2, _N3};
-  static constexpr std::tuple<int32_t, type, int32_t> default_value = {
-      0, type::none, 0};
-};
+// Added in version 3 of this extension.
+#include <sycl/ext/oneapi/latency_control/properties.hpp>
+#include <sycl/ext/oneapi/properties/properties.hpp>
 
 template <typename name,
           typename dataT,
           size_t min_capacity = 0>
 class pipe {
   // Blocking
-  template <class... _Params>
+  template <typename _propertiesT = decltype(sycl::ext::oneapi::experimental::properties{})>
   static dataT read();
```

Comment:
Why have this form where the property list is *only* a template parameter?  If someone wants to specify a property, I presume they would use the other form below, where there is a `Properties` function parameter.  If you just want a way to call `read()` with no properties, why not just define this as a non-template function:

```
static dataT read();
```

Reply:
Because it's a compile-time property list and I need to use `constexpr` to describe the properties.

The purpose of the other "function parameter" form below is only providing users an API to pass the property list as a function parameter. It doesn't process the property list itself, instead, it calls the "template parameter" form inside.

Reply:
The benefit of having it like this is increased freedom. A user could know only the type of the property list and could pass that as a template argument, i.e. `read<propertiesT>()`. That said they should be able to achieve it with `read(propertiesT{})` so it's arguably not a lot you gain.

If we ever think there might be properties with runtime values for this, what @gmlueck suggests would be a lot safer.

Reply:
We don't expect to have any runtime properties at the moment, but @gmlueck's suggestion makes it scalable for future runtime properties (if there will be any), right?
If that's the case, I think there's a value to update the form.

Reply:
>  A user could know only the type of the property list and could pass that as a template argument, i.e. `read<propertiesT>()`

I was wondering if this was the motivation for providing the template form when there is no `Properties` function parameter.  However, I'm not sure we want to encourage code like this.  As you note, this would make it easy to shoot yourself in the foot if we ever added runtime properties.